pkg/scaling/resolver/scale_resolvers.go

```diff
@@ -302,7 +303,7 @@ func resolveEnv(ctx context.Context, client client.Client, logger logr.Logger, c
 				default:
 					return nil, fmt.Errorf("error reading config ref %s on namespace %s/: %w", source.ConfigMapRef, namespace, err)
 				}
-			} else if source.SecretRef != nil && !isSecretAccessRestricted(logger) {
+			} else if source.SecretRef != nil && readSecrets(secretAccessRestricted, namespace) {
```

Comment:
You may want to evaluate `readSecrets(secretAccessRestricted, namespace)` once and assign it to a boolean variable outside of the `for` loop